app/src/main/java/org/example/uiController/UIController.java

```diff
@@ -218,17 +225,48 @@ void handleBatchJobCommand(String[] command) {
             System.out.println("Error: Could not enqueue batch job: " + e.getMessage());
         }
         if (dispatcherThread == null) {
-            dispatcherThread = this.startThread("Dispatcher", dispatcher);
+            dispatcherThread = this.startThread(dispatcher);
+        }
+    }
+
+    /**
+     * Handles the test command when it is submitted by the user.
+     *
+     * @param command The command.
+     */
+    void handleTestCommand(String[] command) {
+        if (command.length != 7) {
+            System.out.println("Invalid test command please try again. \nUsage: test <benchmark> <policy> <number of jobs> <priority level> <min CPU time> <max CPU time>");
+            return;
+        }
+        String benchmarkName = command[1];
+        SchedulingPolicy policy = SchedulingPolicy.valueOf(command[2].toUpperCase());
+        int numJobs = Integer.parseInt(command[3]);
+        int priorityLevels = Integer.parseInt(command[4]);
+        int minCpuTime = Integer.parseInt(command[5]) * 1000; // convert to ms for processing
+        int maxCpuTime = Integer.parseInt(command[6]) * 1000; // convert to ms for processing
+        if (schedulerThread == null) {
+            schedulerThread = this.startThread(scheduler);
+        }
+        if (dispatcherThread == null) {
+            dispatcherThread = this.startThread(dispatcher);
+        }
+        dispatcher.setIsPerfMode(true);
+        PerfTestParams perfTestParams = new PerfTestParams(benchmarkName, policy, numJobs, priorityLevels, minCpuTime, maxCpuTime);
+        PerfEvaluator perfEvaluator = new PerfEvaluator(perfTestParams, scheduler, dispatcher);
+        try {
+            perfEvaluator.run();
+        } catch (InterruptedException e) {
+            System.out.println("Error: Could not run performance evaluation: " + e.getMessage());
```

Comment:
Adding `throws InterruptedException` will remove the IDE errors you encounter when calling a method that could throw an exception, but it won't actually catch the exception. We need to use try/catch blocks for that because otherwise it'll show up in the console when it throws and halt the program.